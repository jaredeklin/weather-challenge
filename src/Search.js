import React, { Component } from 'react';

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      location: ''
    };
  }

  handleSubmit = event => {
    event.preventDefault();
    console.log(this.state.location);
  };

  handleChange = event => {
    const { name, value } = event.target;

    this.setState({
      [name]: value
    });
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input
          type="text"
          name="location"
          placeholder="Enter location"
          onChange={this.handleChange}
          onBlur={this.handleSubmit}
          value={this.state.location}
        />
      </form>
    );
  }
}

export default Search;
