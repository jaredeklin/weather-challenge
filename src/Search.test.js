import React from 'react';
import Search from './Search';
import { shallow } from 'enzyme';

describe('Search', () => {
  let wrapper;
  const mockGetLocationData = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Search getLocationData={mockGetLocationData} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when there is a location in state', () => {
    wrapper.setState({ location: 'Denver' });
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleChange', () => {
    it('should update state with correct values', () => {
      const mockEvent = {
        target: {
          name: 'location',
          value: 'Denver'
        }
      };

      expect(wrapper.state()).toEqual({ location: '' });
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state()).toEqual({ location: 'Denver' });
    });
  });

  describe('handleSubmit', () => {
    it('should call getLoactionData with correct params and reset state', () => {
      const mockEvent = {
        preventDefault: jest.fn()
      };

      wrapper.setState({ location: 'Denver' });
      wrapper.instance().handleSubmit(mockEvent);
      expect(mockGetLocationData).toHaveBeenCalledWith('Denver');
      expect(wrapper.state()).toEqual({ location: '' });
    });
  });
});
