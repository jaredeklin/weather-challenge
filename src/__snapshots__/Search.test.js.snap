// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    getLocationData={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="location"
          onChange={[Function]}
          placeholder="Enter location"
          type="text"
          value=""
        />,
        <button
          disabled={true}
        >
          Submit Location
        </button>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "location",
          "onChange": [Function],
          "placeholder": "Enter location",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit Location",
          "disabled": true,
        },
        "ref": null,
        "rendered": "Submit Location",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="location"
            onChange={[Function]}
            placeholder="Enter location"
            type="text"
            value=""
          />,
          <button
            disabled={true}
          >
            Submit Location
          </button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "location",
            "onChange": [Function],
            "placeholder": "Enter location",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit Location",
            "disabled": true,
          },
          "ref": null,
          "rendered": "Submit Location",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Search should match the snapshot when there is a location in state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    getLocationData={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="location"
          onChange={[Function]}
          placeholder="Enter location"
          type="text"
          value="Denver"
        />,
        <button
          disabled={false}
        >
          Submit Location
        </button>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "location",
          "onChange": [Function],
          "placeholder": "Enter location",
          "type": "text",
          "value": "Denver",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit Location",
          "disabled": false,
        },
        "ref": null,
        "rendered": "Submit Location",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="location"
            onChange={[Function]}
            placeholder="Enter location"
            type="text"
            value="Denver"
          />,
          <button
            disabled={false}
          >
            Submit Location
          </button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "location",
            "onChange": [Function],
            "placeholder": "Enter location",
            "type": "text",
            "value": "Denver",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit Location",
            "disabled": false,
          },
          "ref": null,
          "rendered": "Submit Location",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
